querylog.py

```diff
@@ -121,6 +142,13 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+def emergency_shutdown():
```

Comment:
If I understand correctly, when Python receives SIGKILL (or SIGTERM?), it still waits for operations like this to end gracefully, right?

Reply:
In this specific case it's not Python-specific, it's a `gunicorn` feature. See here: https://docs.gunicorn.org/en/stable/settings.html#worker-exit

The documentation on it is a bit sparse, but by inserting a `time.sleep(40)` in a request handler I've been able to observe that it works and gets called. It doesn't work *exactly* the way I expected -- for example, it seems to be run in a different thread so the thread-local below doesn't actually refer to the in-progress `LogRecord`.

Nevertheless, Flask requests seem to be successfully killed which causes the requests to finish and the LogRecord to be written as well.

We'll have to see how this behaves in production and potentially do something more drastic to track down and write out LogRecords in flight. But for now, maybe this is already enough to get us the information that we need.